### ISSUE TITLE
Update logic for Readme Verification Script

### ISSUE THREAD

**Issue by KarishmaGhiya** [Azure/azure-sdk-tools#1530]
The readme verification script that is being used for verification of readme content for all packages should update it's logic to only include the packages readme and only for track 2 packages and only the readmes at the path :
`sdk\service\package\readme` . Once this issue resolves https://github.com/Azure/autorest.typescript/pull/895 we can use the script to also track track 1 autogenerated packages.

Currently it shows up issues for all the readmes in the repo like the samples readme.md, test readme.md and the swagger readme.md - these readmes will never align with the guidelines, as it doesn't make sense for these to be similar to package readme(s). We should also skip this check for packages that are marked as private in their package.json
Issue reference - https://github.com/Azure/azure-sdk-for-js/issues/1583

**Comment by ramya-rao-a** [comment]
cc @chidozieononiwu, @weshaggard

**Comment by chidozieononiwu** [comment]
I believe this can be easily resolved by adding paths that should be omitted in the [.docsettings.yml](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/.docsettings.yml) file.

**Comment by chidozieononiwu** [comment]
We can probably replace docwarden with an implementation using markdown linter https://github.com/DavidAnson/markdownlint-cli2-action

**Comment by kurtzeborn** [comment]
Another issue talking about the replacement of doc-warden.

**Comment by danieljurek** [comment]
Running the steps in verify-readme.yml locally (in Windows, not Linux) is giving a lot of different output though the steps themselves running in the pipeline don't appear to generate the same output and appear to be succeeding and correctly configured for tests/samples/etc. 

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4133361&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=d34cfe88-f3e1-5ca1-ec50-18a5cea1099a

If there are places where this is failing for track 1 packages those should be added to the excluded files list. 

This issue can be closed as it's not impacting current work and there are ways to unblock problems described.
